File: backend/app/src/main/kotlin/net/animeta/backend/controller/v3/DisconnectTwitterController.kt
package net.animeta.backend.controller.v3

import net.animeta.backend.model.User
import net.animeta.backend.security.CurrentUser
import net.animeta.backend.service.TwitterService
import org.springframework.transaction.annotation.Transactional
import org.springframework.web.bind.annotation.PostMapping
import org.springframework.web.bind.annotation.RestController

@RestController
class DisconnectTwitterController(
    private val twitterService: TwitterService
) {
    data class Result(
        val ok: Boolean
    )

    @PostMapping("/v3/DisconnectTwitter")
    @Transactional
    fun handle(@CurrentUser currentUser: User): Result {
        twitterService.removeOAuthAuthorization(currentUser)
        return Result(true)
    }
}

File: backend-ts/src/controllers/disconnect_twitter.controller.ts
import { Body, Controller, HttpCode, Post } from "@nestjs/common";
import { CurrentUser } from "src/auth/decorators";
import { User } from "src/entities/user.entity";
import { TwitterService } from "src/services/twitter.service";

type Params = {
}

type Result = {
  ok: boolean;
}

@Controller('/api/v4/DisconnectTwitter')
export class DisconnectTwitterController {
  constructor(
    private twitterService: TwitterService,
  ) {}

  @Post()
  @HttpCode(200)
  async handle(
    @Body() params: Params,
    @CurrentUser() currentUser: User,
  ): Promise<Result> {
    await this.twitterService.removeOAuthAuthorization(currentUser)
    return {ok: true}
  }
}
